src/main/java/com/sforce/cd/apexUnit/client/codeCoverage/CodeCoverageComputer.java

```diff
@@ -82,37 +89,100 @@ public CodeCoverageComputer() {
 		 * file and array from regex prefix
 		 */
 		// read class names from manifest files
-		
 		if (CommandLineArguments.getClassManifestFiles() != null) {
 			LOG.debug(" Fetching apex classes from location : " + CommandLineArguments.getClassManifestFiles());
 			classesAsArray = ApexClassFetcherUtils
-					.fetchApexClassesFromManifestFiles(CommandLineArguments.getClassManifestFiles(), true);
+					.fetchApexClassesFromManifestFiles(CommandLineArguments.getClassManifestFiles(),true);
 		}
 		// fetch matching class names based on regex
 		if (CommandLineArguments.getSourceRegex() != null) {
 			LOG.debug(" Fetching apex classes with regex : " + CommandLineArguments.getSourceRegex());
 			classesAsArray = ApexClassFetcherUtils.fetchApexClassesBasedOnMultipleRegexes(connection, classesAsArray,
-					CommandLineArguments.getSourceRegex(), true);
+					CommandLineArguments.getSourceRegex(),true);
 		}
 		// Do not proceed if no class names are returned from both manifest
 		// files and/or regexes
 		if (classesAsArray != null && classesAsArray.length > 0) {
-			String classArrayAsStringForQuery = processClassArrayForQuery(classesAsArray);
-			String relativeServiceURL = "/services/data/v" + SUPPORTED_VERSION + "/tooling";
-			// compute aggregated code coverage
-			String soqlcc = QueryConstructor.getAggregatedCodeCoverage(classArrayAsStringForQuery);
-			
-			JSONObject responseJsonObject = null;
-			responseJsonObject = WebServiceInvoker.doGet(relativeServiceURL, soqlcc, OAuthTokenGenerator.getOrgToken());
-			LOG.debug("responseJsonObject says " + responseJsonObject + "\n relativeServiceURL is " + relativeServiceURL
-					+ "\n soqlcc is " + soqlcc);
-			if (responseJsonObject != null) {
-				apexClassCodeCoverageBeans = processJSONResponseAndConstructCodeCoverageBeans(connection,
-						responseJsonObject);
-			}
-			if (apexClassCodeCoverageBeans == null) {
-				ApexUnitUtils.shutDownWithErrMsg(
-						"Code coverage metrics not computed. Null object returned while processing the JSON response from the Tooling API");
+
+			if (classesAsArray.length > BATCH_SIZE) {
+				// Creating multiple threads for sending request if URL is huge.
+
+				ExecutorService threadPool = Executors.newFixedThreadPool(5);
```

Comment:
@dragon0556 Can you please explain why pool size is chosen to be 5?  Runtime.getRuntime().availableProcessors() should also help in deciding pool size, right?

Reply:
Hi @code-sagar , After discussion with ashwin we came to know that maximum 5 concurrent requests are allowed on the org/sandbox.
Thats why we choose 5 .

Reply:
Runtime.getRuntime().availableProcessors() can give current available processor so it is good when we are working for more than 32 core machines. for lesser it would be additional call.